### PR TITLE
Patch 1

### LINKED ISSUE #59: Installs as apparent version 0.0.0
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

When installing statsmodels on windows with conda, some of the things reporting the version number say 0.0.0.

    PS C:\ProgramData\Miniconda3\pkgs\statsmodels-0.13.5-py39hc266a54_1\Lib\site-packages> cat .\statsmodels-0.0.0.dist-info\METADATA
    Metadata-Version: 2.1
    Name: statsmodels
    Version: 0.0.0
    Summary: Statistical computations and models for Python
    Home-page: https://www.statsmodels.org/
    Download-URL:
    Maintainer: statsmodels Developers
    Maintainer-email: pystatsmodels@googlegroups.com
    ...

This is particularly a problem because I have some `pip`-only dependencies in my environment that also require `statsmodels`, and when they get installed or updated, pip will "upgrade" statsmodels from 0.0.0 to 0.13.5.

### Installed packages

```shell
# packages in environment at C:\ProgramData\Miniconda3\envs\test_statsmodels:
#
# Name                    Version                   Build  Channel
asttokens                 2.1.0              pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
bzip2                     1.0.8                h8ffe710_4    conda-forge
ca-certificates           2022.9.24            h5b45459_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
debugpy                   1.6.3            py39h99910a6_1    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
executing                 1.2.0              pyhd8ed1ab_0    conda-forge
intel-openmp              2022.1.0          h57928b3_3787    conda-forge
ipykernel                 6.17.0             pyh025b116_0    conda-forge
ipython                   8.6.0              pyh08f2357_1    conda-forge
jedi                      0.18.1             pyhd8ed1ab_2    conda-forge
jupyter_client            7.4.4              pyhd8ed1ab_0    conda-forge
jupyter_core              4.11.2           py39hcbf5309_0    conda-forge
libblas                   3.9.0              16_win64_mkl    conda-forge
libcblas                  3.9.0              16_win64_mkl    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
liblapack                 3.9.0              16_win64_mkl    conda-forge
libsodium                 1.0.18               h8d14728_1    conda-forge
libsqlite                 3.39.4               hcfcfb64_0    conda-forge
libzlib                   1.2.13               hcfcfb64_4    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
mkl                       2022.1.0           h6a75c08_874    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
numpy                     1.23.4           py39hbccbffa_1    conda-forge
openssl                   3.0.7                hcfcfb64_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandas                    1.5.1            py39h2ba5b7c_1    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
patsy                     0.5.3              pyhd8ed1ab_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pip                       22.3               pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.32             pyha770c72_0    conda-forge
psutil                    5.9.3            py39ha55989b_1    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pygments                  2.13.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
python                    3.9.13          hcf16a7b_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.9                      2_cp39    conda-forge
pytz                      2022.6             pyhd8ed1ab_0    conda-forge
pywin32                   304              py39h99910a6_2    conda-forge
pyzmq                     24.0.1           py39hea35a22_1    conda-forge
scipy                     1.9.3            py39hfbf2dce_1    conda-forge
setuptools                65.5.0             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.39.4               hcfcfb64_0    conda-forge
stack_data                0.6.0              pyhd8ed1ab_0    conda-forge
statsmodels               0.13.5           py39hc266a54_1    conda-forge
tbb                       2021.6.0             h91493d7_1    conda-forge
tk                        8.6.12               h8ffe710_0    conda-forge
tornado                   6.2              py39ha55989b_1    conda-forge
traitlets                 5.5.0              pyhd8ed1ab_0    conda-forge
tzdata                    2022f                h191b570_0    conda-forge
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
vc                        14.3                 h3d8a991_9    conda-forge
vs2015_runtime            14.32.31332          h1d6e394_9    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
wheel                     0.38.1             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
zeromq                    4.3.4                h0e60522_1    conda-forge
```

### Environment info

```shell
active environment : test_statsmodels
    active env location : C:\ProgramData\Miniconda3\envs\test_statsmodels
            shell level : 1
       user config file : C:\Users\me\.condarc
 populated config files : C:\Users\me\.condarc
          conda version : 22.9.0
    conda-build version : not installed
         python version : 3.9.13.final.0
       virtual packages : __cuda=11.7=0
                          __win=0=0
                          __archspec=1=x86_64
       base environment : C:\ProgramData\Miniconda3  (writable)
      conda av data dir : C:\ProgramData\Miniconda3\etc\conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\ProgramData\Miniconda3\pkgs
                          C:\Users\me\.conda\pkgs
                          C:\Users\me\AppData\Local\conda\conda\pkgs
       envs directories : C:\ProgramData\Miniconda3\envs
                          C:\Users\me\.conda\envs
                          C:\Users\me\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/22.9.0 requests/2.28.1 CPython/3.9.13 Windows/10 Windows/10.0.19044
          administrator : False
             netrc file : None
           offline mode : False
```

### PR DESCRIPTION
**Pull request by theavey**
@ocefpaf proposed fix for #59

possibly fixes #59 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by theavey** [comment]
@conda-forge-admin, please rerender

**Comment by theavey** [comment]
@ocefpaf, doesn't particularly look like that fixed it: [build log](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=605883&view=logs&j=9a864fd9-6c8f-52ca-79ce-2aa6dca1a1de&t=10fc5aa2-324e-5982-4c88-6b31fcab16b3&l=233)

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,17 +9,18 @@ source:
...
 
 requirements:
   build:
     - python                                 # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - cython >=0.29.32                        # [build_platform != target_platform]
+    - cython >=0.29.32                       # [build_platform != target_platform]
     - numpy                                  # [build_platform != target_platform]
     - {{ compiler('c') }}
+    - setuptools_scm[toml] ~=7.0.0
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Conda doesn't understand the extras syntax in Python. We need to be explicitly about the optional dependencies.

```suggestion
    - setuptools_scm ~=7.0.0
    # setuptools_scm[toml]
    - setuptools >=42
    
```

**Comment by ocefpaf** [comment]
We should add a test like `python -c "import statsmodels; print(statsmodels.__version__)"` to check if we have the right version.

File: recipe/meta.yaml (hunk 1 above)
Comment (theavey) [reply recipe/meta.yaml]: yeah, that's easy enough to do, but doesn't help the issue it seems. That [requirement already was satisfied](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=605883&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=124)

**Comment by bashtage** [comment]
It defnitely _can_ work.  It works for linearmodels which also uses setuptools_scm

```
create -n lm -c conda-forge linearmodels python=3.10 -y
conda activate lm
python
>>> import linearmodels
>>> linearmodels.__version__
'4.27'
```

**Comment by theavey** [comment]
> We should add a test like `python -c "import statsmodels; print(statsmodels.__version__)"` to check if we have the right version.

Unfortunately, in this case, it wouldn't be helpful. `statsmodels.__version__` (and `statsmodels/_version.py`) are being set correctly

File: recipe/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipe/meta.yaml]: Fun! Some more context: https://github.com/statsmodels/statsmodels/issues/8497 and https://twitter.com/ocefpaf/status/1562400506437373953

**Comment by bashtage** [comment]
Perhaps remove ` --no-deps` from the recipe

File: recipe/meta.yaml (hunk 1 above)
Comment (bashtage) [reply recipe/meta.yaml]: Move setuptools_scm to host.

**Comment by bashtage** [comment]
Two sugestions

1. Remove --no-deps
2. Move setuptools_scm to host

See https://github.com/conda-forge/linearmodels-feedstock/blob/main/recipe/meta.yaml which works as expected.

File: recipe/meta.yaml (hunk 1 above)
Comment (bashtage) [reply recipe/meta.yaml]: I made these changes.

Comment (bashtage) [reply recipe/meta.yaml]: https://github.com/statsmodels/statsmodels/issues/8497 is unrelated to this.  That was about package managers like poetry that want to know deps without downloading whls.

**Comment by theavey** [comment]
somthing there seemed to fix it

>  creating '/tmp/pip-modern-metadata-9b9on6y9/statsmodels-0.13.5.dist-info'
>  Preparing metadata (pyproject.toml): finished with status 'done'
>  Source in $SRC_DIR has version 0.13.5, which satisfies requirement statsmodels==0.13.5 from file://$SRC_DIR
>  Removed statsmodels==0.13.5 from file://$SRC_DIR from build tracker '/tmp/pip-build-tracker-5t_dh3z1'

**Comment by bashtage** [comment]
It is probably the host section, but not worth finding out IMO.

**Comment by theavey** [comment]
> It is probably the host section, but not worth finding out IMO.

Agree. Doubt it's the `--no-deps`

**Comment by ocefpaf** [comment]
> Agree. Doubt it's the `--no-deps`

Yep. That is always true in conda (and any third party package manager) BTW. I missed that the dependency was not on `host` in my review.

**Pull request merged**